tests/unit/core_tests.py

```diff
@@ -341,3 +346,12 @@ def test_from_upstream_handles_no_sampled_or_flags(self):
         span = TraceInfo.from_upstream(1, 2, 3, None, None)
         self.assertIsNone(span.sampled)
         self.assertIsNone(span.flags)
+
+    def test_from_upstream_handles_no_parent(self):
+        """
+        Verify that `parent_id` can be left out without raising an exception.
+        """
```

Comment:
nit (no need to fix): I would suggest following PEP257 when possible since it produces better output from pydoc (and other tools that parse docstrings): https://peps.python.org/pep-0257/#one-line-docstrings